### PR TITLE
Exclude dates now saving correctly

### LINKED ISSUE #5: Exclude dates not saving properly.
After saving a download, they are then shown as null datetimes (`1970/01/01`).

### PR DESCRIPTION
**Pull request by mecha**
Fixes #5

### CODE REVIEW

File: src/Module/AdminDeleteServiceHandler.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        $postId   = $event->getParam(0);
+        $postType = $this->_getPostType($postId);
+
+        if ($postType === 'download') {
```
Comment (XedinUnknown) [inline src/Module/AdminDeleteServiceHandler.php]: Hard-coded post-type

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

**Review by XedinUnknown — APPROVED** [review]

File: src/Module/AdminEditServiceUiUpdateHandler.php [hunk 2]
```diff
@@ -325,7 +325,15 @@ protected function _processSessionRuleData($serviceId, $ruleData)
             'repeat_until_date'   => strtotime($this->_containerGet($ruleData, 'repeatUntilDate')),
             'repeat_weekly_on'    => implode(',', $this->_containerGet($ruleData, 'repeatWeeklyOn')),
             'repeat_monthly_on'   => implode(',', $this->_containerGet($ruleData, 'repeatMonthlyOn')),
-            'exclude_dates'       => implode(',', $this->_containerGet($ruleData, 'excludeDates')),
         ];
+
+        $excludeDates = [];
+        foreach ($this->_containerGet($ruleData, 'excludeDates') as $_excludeDate) {
+            $excludeDates[] = strtotime(date('Y-m-d', strtotime($_excludeDate)), 0);
```
Comment (XedinUnknown) [inline src/Module/AdminEditServiceUiUpdateHandler.php]: What was the problem, how was it solved? Also, looks like this code could use some abstraction.

Comment (mecha) [reply src/Module/AdminEditServiceUiUpdateHandler.php]: Previously it was saving as date time strings. Now as timestamps, and without the time (i.e. at `00:00:00`).

**Review by XedinUnknown — APPROVED** [review]

File: src/Module/AdminEditServiceUiUpdateHandler.php (hunk 2 above)
Comment (XedinUnknown) [reply src/Module/AdminEditServiceUiUpdateHandler.php]: Aren't we using date-time strings now in the API? And please abstract conversion to a standard format into a method.

Comment (mecha) [reply src/Module/AdminEditServiceUiUpdateHandler.php]: But we don't save date-time strings. So we translate them here for storage into the DB.

I will add abstraction.

**Pull request merged**